geoSegment.js

```diff
@@ -0,0 +1,353 @@
+    superEvent: sessionSeries,
+  };
+
+  // Write ScheduledSession into each output segment directory
+  for (const segmentIdentifier of sessionSeries['imin:segment']) {
+    const filename = `${encodeURIComponent(linkedScheduledSessionData.id)}.json`;
+    await fs.writeJson(`${OUTPUT_DIRECTORY_PATH}/${segmentIdentifier}/${filename}`, linkedScheduledSessionData);
+    await fs.appendFile(`${OUTPUT_DIRECTORY_PATH}/${segmentIdentifier}/index.txt`, `${filename}\r\n`);
+  }
+}
+
+/**
+ * There is a system filepath limit of 255 characters in most file systems.
+ * All sessionseries filepaths with be of the form `./sessionseries/${encodedFilename}.json`
+ * Therefore the encodedFilename has to be less than 234 characters in length (./sessionseries/.json is 21 characters)
+ *
+ * @param {string} sessionSeriesId
+ */
+function getSessionSeriesFilePath(sessionSeriesId) {
+  let encodedFilename = encodeURIComponent(sessionSeriesId);
+  if (encodedFilename.length > 234) {
+    encodedFilename = encodedFilename.slice(0, 234);
+  }
+  return `./${SESSION_SERIES_DIRECTORY_PATH}/${encodedFilename}.json`;
+}
+
+/**
+ *
+ * @param {RpdeItem<SessionSeriesData>[]} items
+ * @param {Segment[]} segments
+ */
+async function processSessionSeriesItems(items, segments) {
+  items.forEach(async (item) => {
+    // Filter deleted SessionSeries
+    if (item.state === 'deleted') {
+      return;
+    }
+
+    // Filter virtual sessions
+    let sessionSeriesPhysicalLocationGeo;
+    if (item.data.location && item.data.location.geo) {
+      sessionSeriesPhysicalLocationGeo = item.data.location.geo;
+    } else if (item.data['beta:affiliatedLocation'] && item.data['beta:affiliatedLocation'].geo) {
+      sessionSeriesPhysicalLocationGeo = item.data['beta:affiliatedLocation'].geo;
+    } else {
+      return;
+    }
+
+    const sessionSeriesData = {
+      ...item.data,
+      'imin:segment': [],
+    };
+
+    // Add imin:segment if applicable
+    for (const segment of segments) {
+      const segmentRadiusInMeters = segment.radius * 1000;
+      const distanceBetweenGeos = geolib.getDistance(
+        { latitude: sessionSeriesPhysicalLocationGeo.latitude, longitude: sessionSeriesPhysicalLocationGeo.longitude },
+        { latitude: segment.latitude, longitude: segment.longitude },
+      );
+      if (distanceBetweenGeos <= segmentRadiusInMeters) {
+        sessionSeriesData['imin:segment'].push(segment.identifier);
+      }
+    }
+    // If there are no segments, drop the SessionSeries as it will not appear in an output folder and therefore we don't need to store it
+    if (sessionSeriesData['imin:segment'].length === 0) {
+      return;
+    }
+
+    // Validate Schedules and generate ScheduledSessions if needed
+    const scheduledSessionsForEveryEventSchedule = [];
+    const correctedEventSchedules = []; // EventSchedules that are type Schedule but are not valid should be changed to PartialSchedule, and not generate ScheduledSessions
```

Comment:
It looks like these aren't used anywhere

Reply:
Corrected